force-app/main/default/classes/SOQL.cls

```diff
@@ -99,6 +130,10 @@ public with sharing class SOQL {
     //[public | private ] [static] data_type method_name (input parameters) {
     //  The body of the method
     //}
+    public static List<Contact> getContactsByEmailDomain () {
+        List<Contact> emailByDomain = [SELECT Name, FirstName, LastName, Email FROM Contact WHERE Email LIKE '%@example.com'];
+        return emailByDomain;
```

Comment:
You'll need to rewrite this method, it doesn't quite do what it is intended to do even if you modified the test to get it to pass.